py/desispec/night_qa.py

```diff
@@ -85,19 +88,40 @@ def get_survey_night_expids(
             )
         )
     )
-    expids, tileids = [], []
+    expids, tileids, surveys = [], [], []
     for i in range(len(fns)):
         hdr = fits.getheader(fns[i], "SPEC")
         if hdr["OBSTYPE"] == "SCIENCE":
-            if hdr["NTSSURVY"] == survey:
-                expids.append(hdr["EXPID"])
-                tileids.append(hdr["TILEID"])
-    log.info(
-        "found {} exposures from {} tiles for SURVEY={} and NIGHT={}".format(
-            len(expids), np.unique(tileids).size, survey, night,
+            survey = "unknown"
+            # AR look for the fiberassign file
+            # AR - used wildcard, because early files (pre-SV1?) were not gzipped
+            # AR - first check SURVEY keyword (should work for SV3 and later)
+            # AR - if not present, take FA_SURV
+            fafns = glob(os.path.join(os.path.dirname(fns[i]), "fiberassign-??????.fits*"))
+            if len(fafns) > 0:
+                fahdr = fits.getheader(fafns[0], 0)
+                if "SURVEY" in [cards[0] for cards in fahdr.cards]:
+                    survey = fahdr["SURVEY"]
+                else:
+                    survey = fahdr["FA_SURV"]
+            if survey == "unknown":
+                log.warning("SURVEY could not be identified for {}; setting to 'unknown'".format(fns[i]))
+            # AR append
+            expids.append(hdr["EXPID"])
+            tileids.append(hdr["TILEID"])
+            surveys.append(survey)
+    expids, tileids, surveys = np.array(expids), np.array(tileids), np.array(surveys)
+    msg = "for NIGHT={} found"
```

Comment:
It looks like `msg` is an unused leftover?

Reply:
correct, nice catch.